libcloud/common/runabove.py

```diff
@@ -55,12 +55,19 @@ class RunAboveResponse(JsonResponse):
     def parse_error(self):
         response = super(RunAboveResponse, self).parse_body()
 
+        status_code = int(self.status)
+        if status_code >= 300 and status_code < 600:
+            raise ProviderError('Unknown error. Status code: %d' % self.status,
+                                http_code=status_code)
         if response.get('errorCode', None) == 'INVALID_SIGNATURE':
```

Comment:
It returns a 400.
For fix: I've put INVALID_SIGNATURE check before status code check.